src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -58,195 +63,147 @@ Device and XCLBIN
+  
+Creating Buffers from the user pointer
+**************************************
 
-        - ``xrt::bo::flags::normal``: Default, Regular Buffer
-        - ``xrt::bo::flags::device_only``: Device only Buffer (meant to be used only by the kernel).
-        - ``xrt::bo::flags::host_only``: Host Only Buffer (buffer resides in the host memory directly transferred to/from the kernel)
-        - ``xrt::bo::flags::p2p``: P2P Buffer, buffer for NVMe transfer  
-        - ``xrt::bo::flags::cacheable``: Cacheable buffer can be used when host CPU frequently accessing the buffer (applicable for embedded platform).
+The ``xrt::bo()`` constructor can also be called using a pointer provided by the user. The user pointer must be aligned to 4K boundary.
 
+.. code:: c++
+      :number-lines: 15
+           
+           // Host Memory pointer aligned to 4K boundary
+           int *host_ptr;
+           posix_memalign(&host_ptr,4096,MAX_LENGTH*sizeof(int)); 
+ 
+           // Sample example filling the allocated host memory       
+           for(int i=0; i<MAX_LENGTH; i++) {
+           host_ptr[i] = i;  // whatever 
+           }
+           
+           auto mybuf = xrt::bo (device, host_ptr, MAX_LENGTH*sizeof(int), kernel.group_id(3)); 
 
 
 2. Data transfer using Buffers
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 XRT Buffer API library provides a rich set of APIs helping the data transfers between the host and the device, between the buffers, etc. We will discuss the following data transfer style
 
-        I. Data transfer between host and device by Buffer read/write API
-        II. Data transfer between host and device by Buffer map API
-        III. Data transfer between buffers by copy API
+I. Data transfer between host and device by Buffer read/write API
+II. Data transfer between host and device by Buffer map API
+III. Data transfer between buffers by copy API
 
 
 I. Data transfer between host and device by Buffer read/write API
 *****************************************************************
 
 To transfer the data from the host to the device, the user first needs to update the host-side buffer backing pointer followed by a DMA transfer to the device. 
 
-The following C APIs are used for the above tasks
-
-    1. ``xrtBOWrite``  
-    2. ``xrtBOSync`` with flag ``XCL_BO_SYNC_BO_TO_DEVICE``
-    
-In C++, ``xrt::bo`` class has following member functions for the same functionality
-
-    1. ``xrt::bo::write``
-    2. ``xrt::bo::sync`` with flag ``XCL_BO_SYNC_BO_TO_DEVICE``
+   
+The ``xrt::bo`` class has following member functions for the same functionality
 
-To transfer the data from the device to the host, the steps are reverse, the user first needs to do a DMA transfer from the device followed by the reading data from the host-side buffer backing pointer. 
+1. ``xrt::bo::write()``
+2. ``xrt::bo::sync()`` with flag ``XCL_BO_SYNC_BO_TO_DEVICE``
```

Comment:
We need a direction flags for this, looks awkward with XCL_*, but that is currently the only way.
Do note, that the write() step is typically not necessary because (1) the BO may have been constructed with a user ptr, or (2) the host buffer of the BO has been accessed using templated map().

Reply:
agreed for new flags instead of XCL_*
Note added for user-pointer